File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/OccupyWorkAreaUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.OccupyWorkAreaCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.WorkAreaResult;
import reactor.core.publisher.Mono;

public interface OccupyWorkAreaUseCase {

    Mono<WorkAreaResult> occupy(OccupyWorkAreaCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/OccupyWorkAreaUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.OccupyWorkAreaCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.WorkAreaResult
import reactor.core.publisher.Mono

interface OccupyWorkAreaUseCase {
    fun occupy(command: OccupyWorkAreaCommand): Mono<WorkAreaResult>
}
